File: jaguar2/Jaguar-Engine/src/com/sinco/bpm/workflow/Token.java
package com.sinco.bpm.workflow;

import com.sinco.bpm.workflow.metadata.Task;
import com.sinco.persistence.Entity;
import com.sinco.persistence.Id;

@Entity
public class Token implements java.io.Serializable {

    private long id;
    
    private Task task;
    
    private ProcessInstance processInstance;
    
    private TokenStatus status;

    @Id
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Task getTask() {
        return task;
    }

    public void setTask(Task task) {
        this.task = task;
    }

    public ProcessInstance getProcessInstance() {
        return processInstance;
    }

    public void setProcessInstance(ProcessInstance processInstance) {
        this.processInstance = processInstance;
    }

    public TokenStatus getStatus() {
        return status;
    }

    public void setStatus(TokenStatus status) {
        this.status = status;
    }
}

File: jaguar/jaguar-server/workflow/engine/token.cpp
#include "token.h"
#include "../metadata/task.h"

long Token::getId() {
    return id;
};

void Token::setId(long _id) {
    id = _id;
};

Task* Token::getTask() {
    return task;
};

void Token::setTask(Task* _task) {
    task = _task;
};

ProcessInstance* Token::getProcessInstance() {
    return processInstance;
};


void Token::setProcessInstance(ProcessInstance* _processInstance) {
    processInstance = _processInstance;
};

TokenStatus Token::getStatus() {
    return status;
};

void Token::setStatus(TokenStatus _status) {
    status = _status;
};

